pkg/humio/runner.go

```diff
@@ -11,10 +11,10 @@ import (
 )
 
 type JobQuerier interface {
-	CreateJob(repo string, query Query) (string, error)
-	DeleteJob(repo string, id string) error
-	PollJob(repo string, id string) (QueryResult, error)
-	ListRepos() ([]string, error)
+	CreateJob(repo string, query Query, authHeaders AuthHeaders) (string, error)
```

Comment:
Instead of passing auth headers to all of these functions, we should have the authHeaders as a member of the struct (Client)
I don't think an interface should describe how something is authenticated. It makes more sense for the implementor or the interface to decide how auth is handled. What do you think?

Reply:
I think my new changes might simplify things 😊